src/Symfony/Component/Dotenv/Command/DebugCommand.php

```diff
@@ -147,36 +149,38 @@ private function getAvailableVars(): array
         return $vars;
     }
 
-    private function getEnvFiles(): array
+    private function getEnvFiles(string $filePath): array
     {
         $files = [
-            '.env.local.php',
-            sprintf('.env.%s.local', $this->kernelEnvironment),
-            sprintf('.env.%s', $this->kernelEnvironment),
+            sprintf('%s.local.php', $filePath),
+            sprintf('%s.%s.local', $filePath, $this->kernelEnvironment),
+            sprintf('%s.%s', $filePath, $this->kernelEnvironment),
         ];
 
         if ('test' !== $this->kernelEnvironment) {
-            $files[] = '.env.local';
+            $files[] = sprintf('%s.local', $filePath);
         }
 
-        if (!is_file($this->getFilePath('.env')) && is_file($this->getFilePath('.env.dist'))) {
-            $files[] = '.env.dist';
+        if (!is_file($filePath) && is_file(sprintf('%s.dist', $filePath))) {
+            $files[] = sprintf('%s.dist', $filePath);
         } else {
-            $files[] = '.env';
+            $files[] = $filePath;
         }
 
         return $files;
     }
 
-    private function getFilePath(string $file): string
+    private function getRelativeName(string $filePath): string
     {
-        return $this->projectDirectory.\DIRECTORY_SEPARATOR.$file;
+        if (str_starts_with($filePath, $this->projectDirectory)) {
+            return substr($filePath, \strlen($this->projectDirectory) + 1);
+        }
+
+        return basename($filePath);
```

Comment:
Only display the basename if the dotenv file is outside of the project directory.